### PR TITLE
143342741 Integrate player search functionality [#143342741]

### PR DESCRIPTION
**Pull request by milesanmi**
#### What does this PR do?
Integrate User search and limit game players to at most 12 people.

#### Description of Task to be completed?
Users should be able to invite people to their game up to a maximum of 11 other people, so no game can exceed 12 people in total and must have at least 3 players.

#### How should this be manually tested?
After cloning the repo, run the repo locally on localhost:3000. Create a new user via the signup link.

Click either the Play Game with Strangers or Play Game with Friends button.

Click on the "Start Game" button to test the limiting functionality and the "Find Users" button to test the User Search functionality.

#### What are the relevant pivotal tracker stories?
#143342741
#### Screenshots (if appropriate)
Image showing User Limiting popup message
screen shot 2017-04-21 at 13 33 21

Image showing User Search functionality
screen shot 2017-04-21 at 13 34 25

### CODE REVIEW

File: config/routes.js [hunk 1]
```diff
@@ -90,4 +90,7 @@ module.exports = function(app, passport, auth) {
     app.get('/play', index.play);
     app.get('/', index.render);
 
+    // EDITTED BY MARANATHA
+    // API routes for user search
+    app.get('/api/search/users/:inviteeSearch?',users.findUsers);
```
Comment (dannijax) [inline config/routes.js]: add space between the two arguments of the function like this `app.get('/api/search/users/:inviteeSearch?', users.findUsers);`

File: public/js/controllers/game.js [hunk 2]
```diff
@@ -90,6 +96,9 @@ angular.module('mean.system')
     };
 
     $scope.isCustomGame = function() {
+      if (game.players.length < 3) {
+        // alert('Minimum of Three player allowed. There are currently ' + game.players.length + ' players. Please Wait A Little!!!');
```
Comment (dannijax) [inline public/js/controllers/game.js]: Remove commented out code if they are no longer needed

File: public/views/question.html [hunk 3]
```diff
@@ -7,18 +75,28 @@
...
       <div id="loading-container">
         <div id="loading-gif"><img ng-src="../img/loader.gif"/></div>
       </div>
-      <div id="start-game-container" ng-click="startGame()" ng-show="(game.playerIndex === 0 || game.joinOverride) && game.players.length >= game.playerMinLimit">
+
+      <!--
+        EDITTED BY MARANATHA
+        THIS CONTROLS THE DISPLAY OF THE "START GAME" BUTTON
+        REMOVING THE ng-show ATTIBUTE DISPLAYS IT. 
+        <div id="start-game-container" ng-click="startGame()" ng-show="(game.playerIndex === 0 || game.joinOverride) && game.players.length >= game.playerMinLimit">
```
Comment (dannijax) [inline public/views/question.html]: remove this line if not needed

File: public/js/services/game.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ angular.module('mean.system')
     table: [],
     czar: null,
     playerMinLimit: 3,
-    playerMaxLimit: 6,
+    playerMaxLimit: 12,
```
Comment (dannijax) [inline public/js/services/game.js:15]: changing a constant value from two different sources is a potential for unpredictable behaviors which mar result in system failure. What if we can save this value in as an environmental variable or constant.js file so we could access it from one source.

File: config/routes.js (hunk 1 above)
Comment (milesanmi) [reply config/routes.js]: Okay. Fixed

File: public/js/controllers/game.js (hunk 2 above)
Comment (milesanmi) [reply public/js/controllers/game.js]: Done.

**Pull request merged**